pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/FunctionsImpl.java

```diff
@@ -419,9 +419,10 @@ public void updateFunction(final String tenant,
 
             updateRequest(functionMetaDataBuilder.build());
         } finally {
-            if (!(functionPkgUrl != null && functionPkgUrl.startsWith(Utils.FILE))
-                    && componentPackageFile != null && componentPackageFile.exists()) {
-                componentPackageFile.delete();
+            if (componentPackageFile != null && componentPackageFile.exists()) {
+                if ((functionPkgUrl != null && !functionPkgUrl.startsWith(Utils.FILE)) || uploadedInputStream != null) {
```

Comment:
Do we also need to exclude "http" ?

Reply:
Do we need to add this for sources and sinks as well?

Reply:
no. Only file. Since for http(or others) we will make a copy and do validation on that copy. We need to cleanup this copy.

Reply:
good catch about sources/sinks. Will add

Reply:
Added the same logic for sources/sinks